_extensions/quarto-ext/shinylive/shinylive.lua

```diff
@@ -77,19 +255,63 @@ function ensureShinyliveSetup(language)
   )
 end
 
+-- Do one-time setup for language specific html dependencies.
+-- This should only be called once per document
+-- @param language: "python" or "r"
+function ensureLanguageSetup(language)
+  ensureInitSetup(language)
+
+  if hasDoneSetup[language] then
+    return
+  end
+  hasDoneSetup[language] = true
+
+  -- Only get the asset version value if it hasn't been retrieved yet.
+  if versions[language] == nil then
+    local infoObj = callShinylive(language, { "extension", "info" })
+    versions[language] = { version = infoObj.version, assets_version = infoObj.assets_version }
+  end
+  -- Verify that the r-shinylive and py-shinylive versions match
+  if
+      (versions.r and versions.python) and
+      ---@diagnostic disable-next-line: undefined-field
+      versions.r.assets_version ~= versions.python.assets_version
+  then
+    error(
+      "The shinylive R and Python packages must support the same Shinylive Assets version to be used in the same quarto document." ..
```

Comment:
- Verify that this works.
- If someone gets this message, there's a good chance they won't understand what the problem is or how to fix it. Maybe something like, "Please update your [whichever of R or Python has older supported assets] package."